### PR TITLE
Improving Exception Handling

### PR DESCRIPTION
**Pull request by BidishaMS**
Improving Exception Handling

1) Adding proper exception name for any API error
2) Previously system error was under API error logging. Now segregated both kind of exceptions

### CODE REVIEW

File: src/web/client/WebExtensionContext.ts [hunk 1]
```diff
@@ -381,14 +384,25 @@ class WebExtensionContext implements IWebExtensionContext {
...
         } catch (error) {
-            const errorMsg = (error as Error)?.message;
-            this.telemetry.sendAPIFailureTelemetry(
-                requestUrl,
-                languageEntityName,
-                Constants.httpMethod.GET,
-                new Date().getTime() - requestSentAtTime,
-                errorMsg
-            );
+            if ((error as Response)?.status>0){
```
Comment (tyaginidhi) [inline src/web/client/WebExtensionContext.ts:387]: nit: spacing between "> 0"

File: src/web/client/WebExtensionContext.ts [hunk 2]
```diff
@@ -381,14 +384,25 @@ class WebExtensionContext implements IWebExtensionContext {
...
+                    requestUrl,
+                    languageEntityName,
+                    Constants.httpMethod.GET,
+                    new Date().getTime() - requestSentAtTime,
+                    errorMsg,
+                    '',
+                    telemetryEventNames.WEB_EXTENSION_POPULATE_LANGUAGE_ID_TO_CODE_API_ERROR,
+                    (error as Response)?.status.toString()
+                );
+            }else{
```
Comment (tyaginidhi) [inline src/web/client/WebExtensionContext.ts:399]: formatting - suggest keeping prettier on

File: src/web/client/dal/remoteFetchProvider.ts [hunk 3]
```diff
@@ -51,7 +50,7 @@ export async function fetchDataFromDataverseAndUpdateVFS(
                 "We encountered an error preparing the file for edit."
             )
         );
-        WebExtensionContext.telemetry.sendErrorTelemetry(telemetryEventNames.WEB_EXTENSION_FAILED_TO_PREPARE_WORKSPACE, errorMsg);
+        WebExtensionContext.telemetry.sendErrorTelemetry(telemetryEventNames.WEB_EXTENSION_FAILED_TO_PREPARE_WORKSPACE, errorMsg,error as Error);
```
Comment (tyaginidhi) [inline src/web/client/dal/remoteFetchProvider.ts:53]: nit: formatting

File: src/web/client/services/etagHandlerService.ts [hunk 4]
```diff
@@ -195,7 +195,7 @@ export class EtagHandlerService {
             } else {
                 WebExtensionContext.telemetry.sendErrorTelemetry(
                     telemetryEventNames.WEB_EXTENSION_ENTITY_CONTENT_UNEXPECTED_RESPONSE,
-                    response.statusText
+                    response.status.toString()
```
Comment (tyaginidhi) [inline src/web/client/services/etagHandlerService.ts:198]: include status text as well

**Review by tyaginidhi — APPROVED** [review]

File: src/web/client/services/etagHandlerService.ts (hunk 4 above)
Comment (BidishaMS) [reply src/web/client/services/etagHandlerService.ts]: fixed

File: src/web/client/dal/remoteFetchProvider.ts (hunk 3 above)
Comment (BidishaMS) [reply src/web/client/dal/remoteFetchProvider.ts:53]: formatting in next PR

File: src/web/client/WebExtensionContext.ts (hunk 2 above)
Comment (BidishaMS) [reply src/web/client/WebExtensionContext.ts:399]: formatting in next PR

**Pull request merged**